### DISCUSSION TITLE
Not being able to access "document" object.

### DISCUSSION THREAD

**Discussion by r1z4x** [alpinejs/alpine#780] [4. General]
Hello there,

I am using more than one nested "x-data" method. But I cannot access the "document" object in the function in x-data. It has a "null" value.

Inside the head tag.

`<script src="https://cdn.jsdelivr.net/gh/alpinejs/alpine@v2.7.0/dist/alpine.min.js" defer></script>
    
    <script src="{{ asset('js/focus-trap.js') }}"></script>
    <script src="{{ asset('js/init-alpine.js') }}"></script>
    
</head>`

Inside HTML "dom" elements.

`<div x-data="editProfileForm()" x-cloak>
                        <button @click="openModal('editProfileModal')" class="text-sm px-3 py-2 font-medium leading-5 text-white transition-colors duration-150 bg-blue-900 rounded-md focus:outline-none focus:shadow-outline-">
                            {{ __('auth.profile.edit_profile') }}
                        </button>
                        <x-modal type="simple" title="{{ __('auth.profile.edit_profile') }}" type="form" modal-id="editProfileModal" form-id="edit-profile-form">
    
                            <form id="edit-profile-form" method="POST" action="/aaaaa">
                                @csrf
                                
    
    
                            </form>
                        </x-modal>
                        <script type="text/javascript">
                            
                            function editProfileForm() {

                                return {
                                    // Modal
                                    isModalOpen: false,
                                    trapCleanup: null,
                                    openModal(id) {
                                        this.isModalOpen = true
                                        this.trapCleanup = focusTrap(document.querySelector('#' + id))
                                    },
                                    closeModal() {
                                        this.isModalOpen = false
                                        this.trapCleanup()
                                    },
                                };

                            }
                        </script>
                    </div>`

**Comment by SimoTod** [comment]
Did you try console.log(document), I tried on different components and it doesn't have a null value. Can you check that null is not the value returned by your querySelector call?

**Reply by r1z4x** [reply]
![image](https://user-images.githubusercontent.com/5313560/93646048-18284580-fa0e-11ea-9416-13798ab8e1b2.png)
![image](https://user-images.githubusercontent.com/5313560/93646082-28402500-fa0e-11ea-9735-b1f335cb555c.png)

It gives the same error I tried. But the DOM ID is in the "template" tag. Could it be caused by it?

![image](https://user-images.githubusercontent.com/5313560/93646405-d350de80-fa0e-11ea-8a2d-97c4ca7c6c0e.png)


But, There is a situation like this. There is an x-data of HTML. It contains general descriptions.
![image](https://user-images.githubusercontent.com/5313560/93646119-44dc5d00-fa0e-11ea-9933-9def730a1131.png)

**Reply by SimoTod** [reply]
Yeah, I think that's the reason (template tag). At that point, alpine hasn't updated the dom yet because the execution is debounced.
You need to use `$nextTick(() => focusTrap(...))` to ensure the modal will be available in the DOM

**Reply by r1z4x** [reply]
The problem has been resolved, thank you.
